### PR TITLE
NMC-552 Move filename/size out of the download button

### PR DESCRIPTION
**Pull request by TSI-sangramdesai**
Guest view below changes done

1. For Video separated download button form file name and size
2. For Image separated download button form file name and size

### CODE REVIEW

**Review by Salvador1313 — COMMENTED** [review]
b20da982f51f9611c2e9288f8f46be9f392ec33d

**Review by Salvador1313 — APPROVED** [review]

**Comment by juliushaertl** [comment]
Please do not include the change in the 3rdparty submodule in here.

In addition the commit is missing a Signed-of-by message as described in https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md#sign-your-work

File: apps/files_sharing/templates/public.php [hunk 1]
```diff
@@ -72,12 +72,28 @@
 				<?php else: ?>
 					<!-- Preview frame is filled via JS to support SVG images for modern browsers -->
 					<div id="imgframe"></div>
+					<?php if(isset($_['mimetype'])){ ?>
+						 <?php	if (strpos($_['mimetype'], 'image') === 0) { ?>
+							<div class="directDownload">
+								<div class="video-file-details">
```
Comment (artonge) [inline apps/files_sharing/templates/public.php:78]: Is the `video-file-details` class useful?

File: apps/files_sharing/templates/public.php [hunk 2]
```diff
@@ -72,12 +72,28 @@
 				<?php else: ?>
 					<!-- Preview frame is filled via JS to support SVG images for modern browsers -->
 					<div id="imgframe"></div>
+					<?php if(isset($_['mimetype'])){ ?>
+						 <?php	if (strpos($_['mimetype'], 'image') === 0) { ?>
```
Comment (artonge) [inline apps/files_sharing/templates/public.php:76]: We can merge those two conditions into one:

```js
isset($_['mimetype']) && strpos($_['mimetype'], 'image') === 0
```

**Review by artonge — CHANGES_REQUESTED** [review]

**Comment by artonge** [comment]
I would also add some space above the `downloadFile` element. Adding the following in `apps/files_sharing/css/public.scss` should be enough:

```css
.directDownload #downloadFile {
	...    
	margin-top: 16px;
}
```

File: apps/files_sharing/templates/public.php (hunk 1 above)
Comment (TSI-sangramdesai) [reply apps/files_sharing/templates/public.php]: Removed

File: apps/files_sharing/templates/public.php (hunk 2 above)
Comment (TSI-sangramdesai) [reply apps/files_sharing/templates/public.php]: Merged

**Review by artonge — APPROVED** [review]
Need DCO, else looks good :)

```bash
git rebase HEAD~6 --signoff
git push --force-with-lease origin NMC-552
```

**Review by skjnldsv — APPROVED** [review]

File: apps/files_sharing/templates/public.php [hunk 3]
```diff
@@ -72,12 +72,26 @@
 				<?php else: ?>
 					<!-- Preview frame is filled via JS to support SVG images for modern browsers -->
 					<div id="imgframe"></div>
+						<?php if (isset($_['mimetype']) && strpos($_['mimetype'], 'image') === 0) { ?>
+							<div class="directDownload">
```
Comment (Pytal) [inline apps/files_sharing/templates/public.php:76]: We should use kebab-case for class and id names :)

Comment (TSI-sangramdesai) [reply apps/files_sharing/templates/public.php:76]: directDownload is original class
https://github.com/nextcloud/server/blob/master/apps/files_sharing/templates/public.php

**Pull request merged**